### PR TITLE
fix: Improve contrast by changing the colors in Start New Conversation View ACC - 220

### PR DESCRIPTION
**Pull request by agisilaos**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

With this PR we update the colors in the Start Conversation View. I've made a couple of changes and in places where I could, I removed the color scheme variant entirely. Slowly but surely we're not going to be dependent on that for our colors in the app.

**Note**: There are a lot of test failures. I'm going to fix them in separate PR's and then merge those PR's in this branch `fix/start-conversation-view-updated-two`. So I think it would be best to review this PR first and then I'm going to start opening the PR's for the test fixes.

----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: Wire-iOS/Sources/Authentication/Interface/ViewControllers/AuthenticationCredentialsViewController.swift [hunk 1]
```diff
@@ -112,7 +112,7 @@ final class AuthenticationCredentialsViewController: AuthenticationStepControlle
         let passwordTab = UITabBarItem(title: "registration.register_by_phone".localized(uppercased: true), image: nil, selectedImage: nil)
         passwordTab.accessibilityIdentifier = "UsePhone"
 
-        return TabBar(items: [emailTab, passwordTab], style: .light)
+        return TabBar(items: [emailTab, passwordTab])
```
Comment (KaterinaWire) [inline Wire-iOS/Sources/Authentication/Interface/ViewControllers/AuthenticationCredentialsViewController.swift:115]: Do we need these changes in this PR? This should not be part of creating a conversation.

File: Wire-iOS/Sources/UserInterface/Components/TabBar/TabBarController.swift [hunk 2]
```diff
@@ -75,7 +75,7 @@ final class TabBarController: UIViewController, UIPageViewControllerDelegate, UI
 
     var style: ColorSchemeVariant = ColorScheme.default.variant {
         didSet {
-            tabBar?.style = style
+
```
Comment (KaterinaWire) [inline Wire-iOS/Sources/UserInterface/Components/TabBar/TabBarController.swift:78]: Do we ned this property?
You can remove `didSet` block

File: Wire-iOS/Sources/UserInterface/Components/Views/UserCell.swift [hunk 3]
```diff
@@ -32,6 +32,7 @@ extension UIImageView {
 class UserCell: SeparatorCollectionViewCell, SectionListCellType {
 
     var hidesSubtitle: Bool = false
+    typealias icon = SemanticColors.IconColors
```
Comment (KaterinaWire) [inline Wire-iOS/Sources/UserInterface/Components/Views/UserCell.swift:35]:
```suggestion
    typealias iconColors = SemanticColors.IconColors
```

File: Wire-iOS/Sources/UserInterface/StartUI/Common/Cells/GroupConversationCell.swift [hunk 4]
```diff
@@ -33,24 +33,9 @@ final class GroupConversationCell: UICollectionViewCell, Themeable {
...
-        didSet {
-            guard oldValue != contentBackgroundColor else { return }
-            applyColorScheme(colorSchemeVariant)
-        }
-    }
-
     override var isHighlighted: Bool {
         didSet {
-            backgroundColor = isHighlighted ? .init(white: 0, alpha: 0.08) : .clear
+            backgroundColor = isHighlighted ? SemanticColors.BackgroundColors.backgroundUserCellHightLighted : SemanticColors.BackgroundColors.backgroundUserCell
```
Comment (KaterinaWire) [inline Wire-iOS/Sources/UserInterface/StartUI/Common/Cells/GroupConversationCell.swift:38]: You can create a `typealias` for `SemanticColors.BackgroundColors`

File: Wire-iOS/Sources/UserInterface/StartUI/StartUI/SearchHeaderViewController.swift [hunk 5]
```diff
@@ -75,7 +75,7 @@ final class SearchHeaderViewController: UIViewController {
 
         tokenField.textView.accessibilityIdentifier = "textViewSearch"
         tokenField.textView.placeholder = "peoplepicker.search_placeholder".localized(uppercased: true)
-        tokenField.textView.keyboardAppearance = ColorScheme.keyboardAppearance(for: colorSchemeVariant)
+        tokenField.textView.keyboardAppearance = .default
```
Comment (KaterinaWire) [inline Wire-iOS/Sources/UserInterface/StartUI/StartUI/SearchHeaderViewController.swift:78]: Does it work for light and dark mode?

File: Wire-iOS/Sources/UserInterface/StartUI/StartUI/StartUIViewController.swift [hunk 6]
```diff
@@ -93,20 +93,16 @@ final class StartUIViewController: UIViewController, SpinnerCapable {
     override func viewDidAppear(_ animated: Bool) {
         super.viewDidAppear(animated)
 
-        navigationController?.navigationBar.barTintColor = UIColor.clear
-        navigationController?.navigationBar.isTranslucent = true
-        navigationController?.navigationBar.tintColor = UIColor.from(scheme: .textForeground, variant: .dark)
+        navigationController?.navigationBar.barTintColor = SemanticColors.BackgroundColors.backgroundViewDefault
+        navigationController?.navigationBar.isTranslucent = false
```
Comment (KaterinaWire) [inline Wire-iOS/Sources/UserInterface/StartUI/StartUI/StartUIViewController.swift:97]: What is the default value? Is it false?

File: Wire-iOS/Sources/Authentication/Interface/ViewControllers/AuthenticationCredentialsViewController.swift (hunk 1 above)
Comment (agisilaos) [reply Wire-iOS/Sources/Authentication/Interface/ViewControllers/AuthenticationCredentialsViewController.swift:115]: Actually it is, since the Tab Bar is part of that view.

File: Wire-iOS/Sources/UserInterface/StartUI/StartUI/SearchHeaderViewController.swift (hunk 5 above)
Comment (agisilaos) [reply Wire-iOS/Sources/UserInterface/StartUI/StartUI/SearchHeaderViewController.swift:78]: Yes!

File: Wire-iOS/Sources/UserInterface/Components/TabBar/TabBarController.swift (hunk 2 above)
Comment (agisilaos) [reply Wire-iOS/Sources/UserInterface/Components/TabBar/TabBarController.swift:78]: Removed it, thanks!

File: Wire-iOS/Sources/UserInterface/Components/Views/UserCell.swift (hunk 3 above)
Comment (agisilaos) [reply Wire-iOS/Sources/UserInterface/Components/Views/UserCell.swift:35]: Done!

File: Wire-iOS/Sources/UserInterface/StartUI/Common/Cells/GroupConversationCell.swift (hunk 4 above)
Comment (agisilaos) [reply Wire-iOS/Sources/UserInterface/StartUI/Common/Cells/GroupConversationCell.swift:38]: Done!

File: Wire-iOS/Sources/UserInterface/StartUI/StartUI/StartUIViewController.swift (hunk 6 above)
Comment (agisilaos) [reply Wire-iOS/Sources/UserInterface/StartUI/StartUI/StartUIViewController.swift:97]: Yes!

File: Wire-iOS/Sources/UserInterface/StartUI/Common/Cells/GroupConversationCell.swift (hunk 4 above)
Comment (KaterinaWire) [reply Wire-iOS/Sources/UserInterface/StartUI/Common/Cells/GroupConversationCell.swift:38]: I don't see the changes

File: Wire-iOS/Sources/UserInterface/StartUI/StartUI/StartUIViewController.swift (hunk 6 above)
Comment (KaterinaWire) [reply Wire-iOS/Sources/UserInterface/StartUI/StartUI/StartUIViewController.swift:97]: Can we remove this line if the default value is `false`?

File: Wire-iOS/Sources/Helpers/SemanticColors.swift [hunk 7]
```diff
@@ -54,6 +54,33 @@ public enum SemanticColors {
         static let searchIcon = UIColor(light: Asset.black, dark: Asset.white)
     }
 
+    public enum BackgroundColors {
```
Comment (KaterinaWire) [inline Wire-iOS/Sources/Helpers/SemanticColors.swift:57]: Will you update semantic colors in this PR?

File: Wire-iOS/Sources/UserInterface/StartUI/Common/Cells/GroupConversationCell.swift (hunk 4 above)
Comment (agisilaos) [reply Wire-iOS/Sources/UserInterface/StartUI/Common/Cells/GroupConversationCell.swift]: Sorry, just pushed them

File: Wire-iOS/Sources/Helpers/SemanticColors.swift (hunk 7 above)
Comment (agisilaos) [reply Wire-iOS/Sources/Helpers/SemanticColors.swift:57]: No, I will do it in a separate PR.

File: Wire-iOS/Sources/UserInterface/Components/Views/UserCell.swift [hunk 8]
```diff
@@ -92,37 +98,55 @@ class UserCell: SeparatorCollectionViewCell, SectionListCellType {
...
             checkmarkIconView.isHidden = true
         }
     }
 
+    override func traitCollectionDidChange(_ previousTraitCollection: UITraitCollection?) {
+        //  Border colors are not dynamically updating for Dark Mode
+        //  When you use adaptive colors with CALayers you’ll notice that these colors,
+        // are not updating when switching appearance live in the app.
+        // That's why we use the traitCollectionDidChange(_:) method.
+        checkmarkIconView.layer.borderColor = iconColors.boarderCellCheckMarkActive.cgColor
```
Comment (johnxnguyen) [inline Wire-iOS/Sources/UserInterface/Components/Views/UserCell.swift:111]: Perhaps it's worth checking that that interface style trait changed, since this method is called for many other unrelated changes.

File: Wire-iOS/Sources/UserInterface/Components/Views/UserCell.swift [hunk 9]
```diff
@@ -32,6 +32,8 @@ extension UIImageView {
 class UserCell: SeparatorCollectionViewCell, SectionListCellType {
 
     var hidesSubtitle: Bool = false
+    typealias iconColors = SemanticColors.IconColors
+    typealias backgroundColors = SemanticColors.BackgroundColors
```
Comment (johnxnguyen) [inline Wire-iOS/Sources/UserInterface/Components/Views/UserCell.swift:36]: Type aliases are still types, so it would be best to name them 'IconColors' and 'BackgroundColors'

File: Wire-iOS/Sources/UserInterface/ConversationList/BackgroundViewController.swift [hunk 10]
```diff
@@ -188,7 +176,6 @@ final class BackgroundViewController: UIViewController {
 
     @objc
     private func colorSchemeChanged() {
-        updateForColorScheme()
     }
```
Comment (johnxnguyen) [inline Wire-iOS/Sources/UserInterface/ConversationList/BackgroundViewController.swift:179]: Any need to keep this method?

File: Wire-iOS/Sources/UserInterface/StartUI/Common/Cells/InviteTeamMemberCell.swift [hunk 11]
```diff
@@ -50,16 +50,21 @@ class StartUIIconCell: UICollectionViewCell {
         createConstraints()
     }
 
+    override func traitCollectionDidChange(_ previousTraitCollection: UITraitCollection?) {
+        iconView.image = icon?.makeImage(size: .tiny, color: SemanticColors.IconColors.foregroundCellIconActive)
+
+    }
```
Comment (johnxnguyen) [inline Wire-iOS/Sources/UserInterface/StartUI/Common/Cells/InviteTeamMemberCell.swift:56]: Again, should we check which traits changed?

File: Wire-iOS/Sources/UserInterface/StartUI/StartUI/SearchResultsView.swift [hunk 12]
```diff
@@ -59,6 +59,10 @@ final class SearchResultsView: UIView {
                                                object: nil)
     }
 
+    override func traitCollectionDidChange(_ previousTraitCollection: UITraitCollection?) {
+        collectionView.reloadData()
+    }
+
```
Comment (johnxnguyen) [inline Wire-iOS/Sources/UserInterface/StartUI/StartUI/SearchResultsView.swift:65]: Why is this needed again? And also we should check which traits changed, otherwise we risk to perform an expensive operation unnecessarily.

**Comment by agisilaos** [comment]
Closing this because #5798 is now merged.
